spanner/spanner_snippets/snippet.go

```diff
@@ -477,6 +483,234 @@ func writeWithTransactionUsingDML(ctx context.Context, w io.Writer, client *span
+
+func pgQueryParameter(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	stmt := spanner.Statement{
+		SQL: `SELECT SingerId, FirstName, LastName FROM Singers
+			WHERE LastName = $1`,
+		Params: map[string]interface{}{
+			"p1": "Garcia",
+		},
+	}
+	type Singers struct {
+		SingerID            int64
+		FirstName, LastName string
+	}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		var val Singers
+		if err := row.ToStruct(&val); err != nil {
+			return err
+		}
+		fmt.Fprintf(w, "%d %s %s\n", val.SingerID, val.FirstName, val.LastName)
+	}
+}
+
+// [END spanner_postgresql_query_parameter]
+
+// [START spanner_postgresql_add_column]
+
+func pgAddNewColumn(ctx context.Context, w io.Writer, adminClient *database.DatabaseAdminClient, database string) error {
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database: database,
+		Statements: []string{
+			"ALTER TABLE Albums ADD COLUMN MarketingBudget bigint",
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(ctx); err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Added MarketingBudget column\n")
+	return nil
+}
+
+// [END spanner_postgresql_add_column]
+
+// [START spanner_postgresql_query_data_with_new_column]
+
+func pgQueryNewColumn(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	stmt := spanner.Statement{SQL: `SELECT SingerId, AlbumId, MarketingBudget FROM Albums`}
+	iter := client.Single().Query(ctx, stmt)
+	defer iter.Stop()
+	for {
+		row, err := iter.Next()
+		if err == iterator.Done {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		var singerID, albumID int64
+		var marketingBudget spanner.NullInt64
+		if err := row.ColumnByName("singerid", &singerID); err != nil {
+			return err
+		}
+		if err := row.ColumnByName("albumid", &albumID); err != nil {
+			return err
+		}
+		if err := row.ColumnByName("marketingbudget", &marketingBudget); err != nil {
+			return err
+		}
+		budget := "NULL"
+		if marketingBudget.Valid {
+			budget = strconv.FormatInt(marketingBudget.Int64, 10)
+		}
+		fmt.Fprintf(w, "%d %d %s\n", singerID, albumID, budget)
+	}
+}
+
+// [END spanner_postgresql_query_data_with_new_column]
+
+// [START spanner_postgresql_dml_getting_started_update]
+
+func pgWriteWithTransactionUsingDML(ctx context.Context, w io.Writer, client *spanner.Client) error {
+	_, err := client.ReadWriteTransaction(ctx, func(ctx context.Context, txn *spanner.ReadWriteTransaction) error {
+		// getBudget returns the budget for a record with a given albumId and singerId.
+		getBudget := func(albumID, singerID int64) (int64, error) {
+			key := spanner.Key{albumID, singerID}
+			row, err := txn.ReadRow(ctx, "Albums", key, []string{"MarketingBudget"})
+			if err != nil {
+				return 0, err
+			}
+			var budget int64
+			if err := row.Column(0, &budget); err != nil {
```

Comment:
Could this just be returned directly instead of checking the error case? Or perhaps (here and elsewhere), we should just add some information to the errors returned? See https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#return-errors